obspy/taup/tau.py

```diff
@@ -11,14 +18,70 @@
 from .taup_time import TauP_Time
 
 
+# From colorbrewer2.org
+COLORS = [
+    "#1f78b4",
+    "#33a02c",
+    "#e31a1c",
+    "#ff7f00",
+    "#6a3d9a",
+    "#a6cee3",
+    "#b2df8a",
+    "#fb9a99",
+    "#fdbf6f",
+    "#cab2d6"]
+
+
+class SmartPolarText(matplotlib.text.Text):
+    """
+    Automatically align text on polar plots to be away from axes.
+
+    This class automatically sets the horizontal and vertical alignments
+    based on which sides of the spherical axes the text is located.
+    """
+
+    def draw(self, renderer, *args, **kwargs):
+        fig = self.get_figure()
+        midx = fig.get_figwidth() * fig.dpi / 2
+        midy = fig.get_figheight() * fig.dpi / 2
+
+        extent = self.get_window_extent(renderer, dpi=fig.dpi)
+        points = extent.get_points()
+
+        is_left = points[0, 0] < midx
+        is_top = points[0, 1] > midy
+        updated = False
+
+        ha = 'right' if is_left else 'left'
+        if self.get_horizontalalignment() != ha:
+            self.set_horizontalalignment(ha)
+            updated = True
+        va = 'bottom' if is_top else 'top'
+        if self.get_verticalalignment() != va:
+            self.set_verticalalignment(va)
+            updated = True
+
+        if updated:
+            self.update_bbox_position_size(renderer)
+
+        matplotlib.text.Text.draw(self, renderer, *args, **kwargs)
+
+
 class Arrivals(list):
     """
-    Class that is returned by the interface methods.
+    List of arrivals returned by the
+    methods of the :class:`~obspy.taup.tau.TauPyModel` class.
+
+    :param arrivals: List of arrivals.
+    :param model: The model used to calculate the arrivals.
+    :param distance: The requested distance.
     """
-    __slots__ = ["arrivals"]
+    __slots__ = ["model", "distance"]
```

Comment:
So is it necessary to hold on to the distance here? Could you use `arrival.distance` vs. `arrival.purist_distance` instead? That way, the multiple arrival scenario in the tutorial could just be `+=` and a single `plot`.